### PR TITLE
Add workspace-enabled example contracts

### PR DESCRIPTION
**Pull request by aon**
This PR creates a workspace where dependencies are defined and several smart-contracts as part of the same workspace, with inherited dependencies.

It was created following the recommendation made in https://github.com/paritytech/cargo-contract/pull/1358. This should be merged after the mentioned PR is merged, given I had to force the branch version for running `cargo-contract` in the CI.

### CODE REVIEW

**Comment by ascjones** [comment]
https://github.com/paritytech/cargo-contract/pull/1358 is merged now, please update this branch

File: workspace-contracts/erc20/lib.rs [hunk 1]
```diff
@@ -0,0 +1,675 @@
...
+                from: Some(*from),
+                to: Some(*to),
+                value,
+            });
+            Ok(())
+        }
+    }
+
+    #[cfg(test)]
+    mod tests {
```
Comment (ascjones) [inline workspace-contracts/erc20/lib.rs:218]: Could we remove these and other tests? Reason is if the ink testing API changes then these will just be extra tests to update, they are just duplicates of existing ones anyway

**Review by ascjones — COMMENTED** [review]
Could we use some dumb contracts instead like `contract1`, `contract2`, `contract3`, with e.g. just a single message and constructor. We only need to demonstrate that they compile as part of a workspace. 

It just means that when the original example e.g. `erc20` changes we don't need to remember to update the copy of it here too.

File: workspace-contracts/erc20/lib.rs (hunk 1 above)
Comment (tenuki) [reply workspace-contracts/erc20/lib.rs:218]: Understood. I'll recheck / rework on this. Probably re-create the PR. Thanks!

Comment (tenuki) [reply workspace-contracts/erc20/lib.rs:218]: (Also trying to adapt the merge to the actual HEAD of main version which changed the execution matrix)

Comment (tenuki) [reply workspace-contracts/erc20/lib.rs:218]: I'll be creating a new PR to replace this one from a new temporary and updated repo, fixing this things. Sorry for the inconvenience. I'll update this PR with a link to the new-one.

**Comment by tenuki** [comment]
This can be closed in favor of PR:52 : https://github.com/paritytech/ink-examples/pull/52
